### PR TITLE
docs: reorg toc and add first draft hubsearch

### PR DESCRIPTION
**Pull request by dbouwman**
1. Description:

- re-organize guides toc
- write Hub Search Guide
- update many other guides
- add better doc comments for a number of interfaces

TOC re-org 
image

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

File: docs/src/guides/concepts/hub-search.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+| `item`  | `tags`, `type`, `access`, `contentstatus`, `categories` |
+| `user`  | none                                                    |
+| `group` | none                                                    |
+
+Once the search is executed, the aggregations are available on the `IHubSearchResponse.aggregations` property as an array of `IHubAggregation` objects.
+
+Applications can use aggregations to construct dynamic facets and filters.
+
+```json
+// Example aggregation stucture
```
Comment (sonofflynn89) [inline docs/src/guides/concepts/hub-search.md:56]: If you don't have an example structure ready, could you add a todo?

File: packages/common/src/search/types/IHubSearchOptions.ts [hunk 2]
```diff
@@ -35,23 +35,24 @@ export interface IHubSearchOptions {
...
+   * DEPRECATED: Use `include` instead
    */
   enrichments?: Enrichments[];
   /**
    * Maximum number of results to return, per-page
    */
   num?: number;
   /**
-   * What page of results to return
+   * LEGACY. Use `start` and `num` instead
```
Comment (sonofflynn89) [inline packages/common/src/search/types/IHubSearchOptions.ts:55]: For my edification, what's the difference between `legacy` and `deprecated` in this context.

**Review by sonofflynn89 — APPROVED** [review]
LGTM! I really appreciated your examples. Just a couple comments 👍

File: packages/common/src/search/types/IHubSearchOptions.ts (hunk 2 above)
Comment (dbouwman) [reply packages/common/src/search/types/IHubSearchOptions.ts:55]: Good question - until we retire the v3 api, we can't kill LEGACY things. At least was my thinking at the time I wrote this

File: docs/src/guides/concepts/hub-search.md (hunk 1 above)
Comment (dbouwman) [reply docs/src/guides/concepts/hub-search.md:56]: I'll be adding to this in the next week

**Pull request merged**